api/v1alpha1/applicationvolumereplication_types.go

```diff
@@ -17,34 +17,42 @@ limitations under the License.
 package v1alpha1
 
 import (
+	plrv1 "github.com/open-cluster-management/multicloud-operators-placementrule/pkg/apis/apps/v1"
+	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// Action that will be taken by Ramen when a subscription is paused
+// DRAction which will be either a failover or failback action
 // +kubebuilder:validation:Enum=Failover;Failback
-type Action string
+type DRAction string
 
-// These are the valid values for Action
+// These are the valid values for DRAction
 const (
-	// Failover, restore PVs to the new home cluster, and unpause subscription
-	ActionFailover Action = "Failover"
+	// Failover, restore PVs to the TargetCluster
+	ActionFailover DRAction = "Failover"
 
-	// Failback, restore PVs to the original home cluster, and unpause subscription
-	ActionFailback Action = "Failback"
-)
+	// Failback, restore PVs to the PreferredCluster
+	ActionFailback DRAction = "Failback"
 
-// DREnabledSubscriptionsMap defines the action used for failover per subscription.
-// Key is subscription name. Value is either empty, 'failover', or 'failback'
-type DREnabledSubscriptionsMap map[string]Action
+	// Relocate, restore PVs to the designated TargetCluster.  PreferredCluster will change
+	// to be the TargetCluster.
+	ActionRelocate DRAction = "Relocate"
+)
 
 // ApplicationVolumeReplicationSpec defines the desired state of ApplicationVolumeReplication
 type ApplicationVolumeReplicationSpec struct {
-	// DREnabledSubscriptions holds Subscription name as the key and an Action as the value
-	DREnabledSubscriptions DREnabledSubscriptionsMap `json:"drEnabledSubscriptions,omitempty"`
+	// Placement is used by AVR
+	Placement *plrv1.Placement `json:"placement"`
```

Comment:
This should be of type v1.ObjectReference as we want a reference to an existing PlRule. Current format allows a user to create a "placement"  with "GenericPlacementFields" for example as part of AVR itself.

Reply:
This is the same construct the ACM Subscription uses.  I borrowed it from there.  I see no reason not to use v1.ObjectReference.

Reply:
Done